design-documents/dependencies-injected-in-properties.md

```diff
@@ -0,0 +1,34 @@
+## Introduce the possibility to inject dependencies directly to properties
+### Reasons
+* Allowing backward compatible introduction of new dependencies to classes without having to use ObjectManager directly  
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Currently we have to add optional arguments to classes and invoke ObjectManager::getInstance() to get a dependency
+for cases when the constructor is invoked from a child class. Declaring such dependencies for properties will allow us to avoid this.
+* Allowing circular dependencies  
```

Comment:
This contradicts with [Magento Technical Guidelines](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html)
3.1. There SHOULD be no circular dependencies between object.

Reply:
Ok, I'll remove this part